src/pages/home/report/ReportActionItem.js

```diff
@@ -122,7 +125,7 @@ class ReportActionItem extends Component {
                             {this.props.shouldDisplayNewIndicator && (
                                 <UnreadActionIndicator />
                             )}
-                            <View style={getReportActionItemStyle(hovered)}>
+                            <View style={getReportActionItemStyle(hovered)} onLayout={this.props.onLayout}>
```

Comment:
Oh nice, didn't know that View has an `onLayout` prop.

Reply:
https://reactnative.dev/docs/view#onlayout

The docs do say

>This event is fired immediately once the layout has been calculated, but the new layout may not yet be reflected on the screen at the time the event is received, especially if a layout animation is in progress.

I think this is maybe as close as we can get to treating an individual chat message as "rendered". There might be a better indicator to use.

Reply:
Hey I thought that as well, I don't mean to get you sidetracked but here's what I've found looking for an alternative:

[viewabilityConfigCallbackPairs](https://reactnative.dev/docs/virtualizedlist#viewabilityconfigcallbackpairs)
> List of ViewabilityConfig/onViewableItemsChanged pairs. A specific onViewableItemsChanged will be called when its corresponding ViewabilityConfig's conditions are met. See ViewabilityHelper.js for flow type and further documentation.

And in [ViewabilityHelper.js](https://github.com/facebook/react-native/blob/dc80b2dcb52fadec6a573a9dd1824393f8c29fdc/Libraries/Lists/ViewabilityHelper.js#L34-L40)
> Minimum amount of time (in milliseconds) that an item must be physically viewable before the
viewability callback will be fired.

I've never used this but it seems you can configure this and wait for the callback to get called for all the items or just the first 5-10 (Thought what about when there are less than 10...)

Reply:
Anyway I think `onLayout` is close enough maybe 1-2 frames before the item is rendered so that like 16-32ms which is probably too little to care about

Reply:
Thanks, that's interesting! The docs are kind of sparse on that stuff so I'm not too sure if I'll go down this road right now. I think it will maybe be slightly more complicated to use this then to do the naive idea I had.

Reply:
Mostly because `onViewableItemsChanged` is called not just when the items first appear, but when things scroll etc. And I don't want to think about it 😄